fakenet/diverters/diverterbase.py

```diff
@@ -1782,3 +1800,68 @@ def maybeExecuteCmd(self, pkt, pid, comm):
             self.logger.info('Executing command: %s' % (execCmd))
             self.execute_detached(execCmd)
 
+    def mapProxySportToOrigSport(self, orig_sport, proxy_sport, is_ssl_encrypted):
+        """Maps Proxy initiated source ports to their original source ports.
+
+        The Proxy listener uses this method to notify the diverter
+        about the proxy originated source port for the original
+        source port. It also notifies if the packet uses SSL
+        encryption.
+
+        Args:
+            orig_sport: int source port that originated the packet
+            proxy_sport: int source port initiated by Proxy listener
+            is_ssl_encrypted: bool is the packet SSL encrypted or not
+
+        Returns:
+            None
+        """
+        self.proxy_orig_sports[proxy_sport] = orig_sport
+        self.proxy_orig_sports_ssl_encrypted[(orig_sport, proxy_sport)] = is_ssl_encrypted
+
+    def logNbi(self, listener_port, nbi, appl_layer_proto, is_ssl_encrypted):
+        """Collects the NBIs from all listeners into a dictionary.
+
+        All listeners (currently only HTTPListener) use this
+        method to notify the diverter about any NBI captured
+        within their scope.
+
+        Args:
+            listener_port: int port bound by listener
+            nbi: dict NBI captured within the listener
+            appl_layer_proto: str Application layer protocol of the pkt
+            is_ssl_encrpted: str is the listener configured to use SSL or not
+
+        Returns:
+            None
+        """
+        proxied_nbi = listener_port in self.proxy_orig_sports
+        orig_sport = (self.proxy_orig_sports[listener_port]
+                        if proxied_nbi else 
+                        listener_port)
+        _, __, pid, comm, orig_dport, proto = self.sessions.get(orig_sport)
+
+        # For proxied nbis, override the listener's is_ssl_encrypted
+        # with Proxy listener's is_ssl_encrypted.
+        # For non-proxied nbis, use listener provided is_ssl_encrypted
+        if proxied_nbi:
+            is_ssl_encrypted = (
```

Comment:
don't need the parenthesis here

Reply:
you could also move this up to the above `if proxied_nbi:` block

Reply:
> you could also move this up to the above `if proxied_nbi:` block

Sorry, not clear which code you are pointing to, could you mention the lines please?

Reply:
Got the point now!